src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1384,7 +987,7 @@ public long getDistanceSquared(@NotNull final BlockPos pos)
     @Override
     public boolean hasTownHall()
     {
-        return townHall != null;
+        return buildingManager.hasTownHall();
```

Comment:
Why do these method still exists?

Reply:
Because it's part of IColony and called on client and serverSide